src/commands/delete.ts

```diff
@@ -53,26 +53,26 @@ function getItemToDelete(contextItem: TektonNode, selectedItems: TektonNode[]):
 
 async function doDelete(items: TektonNode[], toRefresh: Refreshable, commandId?: string): Promise<void | string> {
   if (items) {
-    const taskRunList = await getTaskRunResourceList();
     let message: string;
-    let refResource = true;
+    let hasAnyReference = true;
+    const pipelineList = await getPipelineList();
```

Comment:
Here you should check if there is atleast a task or clustertask in the items you are going to delete otherwise we don't need the pipelineList at all. You can also use the `hasAnyReference` variable above or create a new one, your choice

Something like
```
hasAnyTask = items.some(x => x.context === task || ct)
if (checkRefResource() && hasAnyTask) {
   getpipelinelist
}
```